Realm/Realm/Handles/SessionHandle.cs

```diff
@@ -323,48 +330,67 @@ private static void HandleSessionError(IntPtr sessionHandlePtr, ErrorCode errorC
         [MonoPInvokeCallback(typeof(NativeMethods.NotifyBeforeClientReset))]
         private static bool NotifyBeforeClientReset(IntPtr beforeFrozen, IntPtr managedSyncConfigurationHandle)
         {
+            ClientResetHandlerBase.BeforeResetCallback cb = null;
+            SyncConfigurationBase syncConfig = null;
+
             try
             {
                 var syncConfigHandle = GCHandle.FromIntPtr(managedSyncConfigurationHandle);
-                var syncConfig = (SyncConfigurationBase)syncConfigHandle.Target;
+                syncConfig = (SyncConfigurationBase)syncConfigHandle.Target;
+                cb = syncConfig.ClientResetHandler switch
+                {
+                    DiscardUnsyncedChangesHandler handler => handler.OnBeforeReset,
```

Comment:
What if it's `DiscardLocalResetHandler`? Even though the type is obsolete, we should not break users who were using it.